api/src/test/java/org/bahmni/module/communication/service/impl/MessageBuilderImplTest.java

```diff
@@ -0,0 +1,163 @@
+                .thenReturn(expectedTemplate);
+
+        String result = messageBuilderService.getAppointmentReminderMessage(arguments, providers);
+
+        String expectedResult = expectedTemplate + "nullnullnull";
+        assertEquals(expectedResult, result);
+    }
+
+    @Test
+    public void testGenerateAppointmentMessage() {
+        Map<String, String> arguments = new HashMap<>();
+        arguments.put("key", "value");
+        arguments.put("appointmentKind", "Virtual");
+        List<String> providers = Collections.singletonList("Provider1");
+        String smsTemplate = "Appointment SMS Template";
+        String expectedTemplate = "Generated Appointment SMS Template";
+
+        when(smsTemplateService.templateMessage(smsTemplate, arguments)).thenReturn(expectedTemplate);
+        when(smsTemplateService.templateMessage(MessageBuilderServiceImpl.APPOINTMENT_PROVIDER_TEMPLATE, arguments))
+                .thenReturn("Provider Template");
+        when(smsTemplateService.templateMessage(MessageBuilderServiceImpl.APPOINTMENT_TELECONSULTATION_LINK_TEMPLATE, arguments))
+                .thenReturn("Teleconsultation Link Template");
+        when(smsTemplateService.templateMessage(MessageBuilderServiceImpl.HELPDESK_TEMPLATE, arguments))
+                .thenReturn("Helpdesk Template");
+
+        String result = messageBuilderService.generateAppointmentMessage(arguments, providers, smsTemplate);
+
+        assertEquals("Generated Appointment SMS TemplateProvider TemplateTeleconsultation Link TemplateHelpdesk Template", result);
+    }
+
+    @Test
+    public void testGenerateAppointmentMessageWhenProviderIsEmpty() {
+        Map<String, String> arguments = new HashMap<>();
+        arguments.put("key", "value");
+        arguments.put("appointmentKind", "Virtual");
+        List<String> providers = new ArrayList<>();
+        String smsTemplate = "Appointment SMS Template";
+        String expectedTemplate = "Generated Appointment SMS Template";
+
+        when(smsTemplateService.templateMessage(smsTemplate, arguments)).thenReturn(expectedTemplate);
+        when(smsTemplateService.templateMessage(MessageBuilderServiceImpl.APPOINTMENT_TELECONSULTATION_LINK_TEMPLATE, arguments))
+                .thenReturn("Teleconsultation Link Template");
+        when(smsTemplateService.templateMessage(MessageBuilderServiceImpl.HELPDESK_TEMPLATE, arguments))
+                .thenReturn("Helpdesk Template");
+
+        String result = messageBuilderService.generateAppointmentMessage(arguments, providers, smsTemplate);
+
+        assertEquals("Generated Appointment SMS TemplateTeleconsultation Link TemplateHelpdesk Template", result);
+    }
+
+    @Test
+    public void testGenerateAppointmentMessageWhenAppointmentKindIsNotVirtual() {
+        Map<String, String> arguments = new HashMap<>();
+        arguments.put("key", "value");
+        arguments.put("appointmentKind", "InPerson");
```

Comment:
Is this same as earlier tests, with only appointmentKind being different? If so, what is the difference expected in output email? The test seems to indicate that output email is the same as the one in Virtual appointments?

Reply:
I'll refactor the test to better indicate the expectation when appointmentKind is virtual.